src/main/java/io/github/arrayv/sortdata/SortInfo.java

```diff
@@ -34,53 +40,105 @@ private SortInfo(int id, SortInfo sort) {
         this.runName = sort.runName;
         this.runAllName = sort.runAllName;
         this.category = sort.category;
-        this.slowSort = sort.slowSort;
         this.bogoSort = sort.bogoSort;
         this.radixSort = sort.radixSort;
         this.bucketSort = sort.bucketSort;
+        this.question = sort.question;
+        this.defaultAnswer = sort.defaultAnswer;
+        this.answerValidator = sort.answerValidator;
         this.fromExtra = sort.fromExtra;
     }
 
+    @SuppressWarnings("deprecation")
     public SortInfo(int id, Class<? extends Sort> sortClass) {
         this.id = id;
         this.internalName = sortClass.getSimpleName();
         try {
             this.instanceSupplier = new NewSortInstance(sortClass);
+            this.answerValidator = new MethodAnswerValidator(sortClass);
         } catch (NoSuchMethodException | IllegalAccessException e) {
-            throw new RuntimeException(e);
-        }
-        Sort sort = getFreshInstance();
-        this.disabled = !sort.isSortEnabled();
-        this.unreasonableLimit = sort.getUnreasonableLimit();
-        this.listName = sort.getSortListName();
-        this.runName = sort.getRunSortName();
-        this.runAllName = sort.getRunAllSortsName();
-        this.category = sort.getCategory();
-        this.slowSort = sort.isUnreasonablySlow();
-        this.bogoSort = sort.isBogoSort();
-        this.radixSort = sort.isRadixSort();
-        this.bucketSort = sort.usesBuckets();
+            throw new Error(e);
```

Comment:
Because it shouldn't reasonably happen.
> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch. Most such errors are abnormal conditions. The ThreadDeath error, though a "normal" condition, is also a subclass of Error because most applications should not try to catch it.

Reply:
I don't feel it's serious or abnormal enough to be an `Error`, and it's analogous to invalid arguments to a constructor, but I suppose it still makes little sense to happen, so up to you

Reply:
This code can actually never be reached, hence the `Error`.

Reply:
then why not use an `AssertionError`?

Reply:
Because sometimes I just don't. No real reason.

Reply:
...so?

Reply:
So I should probably change it.